src/libraries/System.Net.Security/src/System/Net/Security/SslStreamCertificateContext.cs

```diff
@@ -109,12 +114,19 @@ public static SslStreamCertificateContext Create(X509Certificate2 target, X509Ce
             // On Linux, AddRootCertificate will start a background download of an OCSP response,
             // unless this context was built "offline", or this came from the internal Create(X509Certificate2)
             ctx.SetNoOcspFetch(offline || noOcspFetch);
-            ctx.AddRootCertificate(root);
+
+            bool transferredOwnership = false;
+            ctx.AddRootCertificate(root, ref transferredOwnership);
+
+            if (!transferredOwnership)
+            {
+                root?.Dispose();
```

Comment:
should we dispose it in finalizer if we do not do it here?

Reply:
I don't think so. The safe handle backing the certificate will have its own finalizer that'll clean up native resources.